client/js/app/actions/ExplorerActions.js

```diff
@@ -217,7 +217,8 @@ var ExplorerActions = {
     updates.loading = false;
 
     if (!ExplorerUtils.responseSupportsChartType(response, explorer.metadata.visualization.chart_type, explorer.query.analysis_type)) {
-      updates.metadata.visualization.chart_type = ExplorerUtils.getChartTypeOptions(response, explorer.query.analysis_type)[0];
+      updates.metadata = _.cloneDeep(explorer.metadata);
+      updates.metadata.visualization.chart_type = ExplorerUtils.getChartTypeOptions(response.query)[0];
```

Comment:
Looks like the `singular` data type addition fixed this. Can you take another look when you get a moment?